### PR TITLE
Verify that the pgrx active thread is actually the main thread

### PR DESCRIPTION
**Pull request by thomcc**
Unrelated to my other work, but this has bugged me for a while and I figured I'd fix it.

### CODE REVIEW

File: pgrx-pg-sys/src/submodules/thread_check.rs [hunk 1]
```diff
@@ -38,8 +40,60 @@ pub(crate) fn check_active_thread() {
...
+    #[cfg(any(target_os = "macos", target_os = "openbsd", target_os = "freebsd"))]
+    return unsafe {
+        match libc::pthread_main_np() {
+            1 => Some(true),
+            0 => Some(false),
+            // Note that this returns `-1` in some error conditions.
+            //
+            // In these cases we are almost certainly not the main thread, but
+            // we don't know -- it's better for this function to return `None`
+            // in cases of uncertainty.
```
Comment (workingjubilee) [inline pgrx-pg-sys/src/submodules/thread_check.rs:60]: They claim this is about the thread not finishing initialization? https://opensource.apple.com/source/libpthread/libpthread-218.1.3/man/pthread_main_np.3.auto.html

File: pgrx-pg-sys/src/submodules/thread_check.rs [hunk 2]
```diff
@@ -10,8 +10,10 @@
 //! Enforces thread-safety in `pgrx`.
 //!
 //! It's UB to call into Postgres functions from multiple threads. We handle
-//! this by remembering the first thread to call into postgres functions, and
-//! panicking if we're called from a different thread.
+//! this by remembering the first thread to call into postgres functions (the
+//! "active thread" in some comments), and panicking if we're called from a
```
Comment (workingjubilee) [inline pgrx-pg-sys/src/submodules/thread_check.rs:14]: and also the big `static ACTIVE_THREAD`.... ^^;

**Review by workingjubilee — APPROVED** [review]

**Comment by workingjubilee** [comment]
cc @LuminousMonkey By the way, is there something more clever we could be doing here for illumos?

**Comment by LuminousMonkey** [comment]
@workingjubilee Sorry, I'm not sure to be honest. I only knew the one change I had to do so I could complete it on a Illumos machine. My working knowledge of things like this is minimal at best.

**Pull request merged**